libs/velo-external-db-core/src/service/data.ts

```diff
@@ -71,8 +72,8 @@ export default class DataService {
 
 
     // sort, skip, limit are not really optional, after we'll implement in all the data providers we can remove the ?
-    async aggregate(collectionName: string, filter: Filter, aggregation: Aggregation, sort?: Sort[], skip?: number, limit?: number) {
-        const totalCount = this.storage.count(collectionName, filter)
+    async aggregate(collectionName: string, filter: Filter, aggregation: Aggregation, sort?: Sort[], skip?: number, limit?: number, returnTotalCount?: boolean) {
+        const totalCount = returnTotalCount ? this.storage.count(collectionName, filter) : undefined
```

Comment:
maybe extract this line to private method? besides that it appears twice, I think it would be more readable that way.